opentelemetry-api/src/opentelemetry/opentelemetry.py

```diff
@@ -0,0 +1,67 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from inspect import signature
+
+
+class OpenTelemetry:
+    def __init__(self, *args, **kwargs) -> None:
+
+        args = list(args)
+
+        object.__setattr__(self, "_repr", [])
+
+        parameters = signature(self.__init__).parameters.values()
```

Comment:
since this is now called for every created OpenTelemetry object is there any negative impact on performance?
e.g. creating a new SDK span would execute this code for:
- Tracer:
  * sampling -> SamplingResult
  * SpanContext
  * _Span
- Span:
  * attributes: `BoundedAttributes`
  * events:  `BoundedList`
    * attributes (per event): `BoundedAttributes`
  * links: `BoundedList`
    * attributes (per link): `BoundedAttributes`

Reply:
Good point, made this method lazy.